Libraries/LibWeb/IndexedDB/IDBObjectStore.cpp

```diff
@@ -222,4 +222,101 @@ WebIDL::ExceptionOr<void> IDBObjectStore::delete_index(String const& name)
+    // 8. If key was given, then:
+    if (key_was_given) {
+        // 1. Let r be the result of converting a value to a key with key. Rethrow any exceptions.
+        auto maybe_key = convert_a_value_to_a_key(realm, key.value());
+        // 2. If r is invalid, throw a "DataError" DOMException.
+        if (maybe_key.is_error())
+            return WebIDL::DataError::create(realm, "Key is invalid"_string);
+
+        // 3. Let key be r.
+        key_value = maybe_key.release_value();
+    }
+
+    // 9. Let targetRealm be a user-agent defined Realm.
+    auto& target_realm = realm;
+
+    // 10. Let clone be a clone of value in targetRealm during transaction. Rethrow any exceptions.
+    auto clone = TRY(clone_in_realm(target_realm, value, transaction));
+
+    // 11. If store uses in-line keys, then:
+    if (store.uses_inline_keys()) {
+        // 1. Let kpk be the result of extracting a key from a value using a key path with clone and store’s key path. Rethrow any exceptions.
+        auto kpk = extract_a_key_from_a_value_using_a_key_path(realm, clone, store.key_path().value());
+
+        // 2. If kpk is invalid, throw a "DataError" DOMException.
+        if (kpk.is_error())
+            return WebIDL::DataError::create(realm, "Key path is invalid"_string);
+
+        auto maybe_kpk = kpk.release_value();
+
+        // 3. If kpk is not failure, let key be kpk.
+        if (!maybe_kpk.is_error()) {
+            key_value = maybe_kpk.release_value();
+        }
+
+        // 4. Otherwise (kpk is failure):
+        else {
+            // 1. If store does not have a key generator, throw a "DataError" DOMException.
+            if (!store.key_generator().has_value())
+                return WebIDL::DataError::create(realm, "Store does not have a key generator"_string);
+
+            // 2. Otherwise, if check that a key could be injected into a value with clone and store’s key path return false, throw a "DataError" DOMException.
+            if (!check_that_a_key_could_be_injected_into_a_value(realm, clone, store.key_path().value()))
+                return WebIDL::DataError::create(realm, "Key could not be injected into value"_string);
+        }
+    }
+
+    // 12. Let operation be an algorithm to run store a record into an object store with store, clone, key, and no-overwrite flag.
+    auto operation = GC::Function<WebIDL::ExceptionOr<JS::Value>()>::create(realm.heap(), [&realm, &store, clone, key_value, no_overwrite] {
+        auto maybe_key = store_a_record_into_an_object_store(realm, store, clone, key_value, no_overwrite);
+        if (maybe_key.is_error())
+            return WebIDL::ExceptionOr<JS::Value>(maybe_key.release_error());
+
+        auto optional_key = maybe_key.release_value();
+        if (optional_key == nullptr)
+            return WebIDL::ExceptionOr<JS::Value>(JS::js_undefined());
+
+        return WebIDL::ExceptionOr<JS::Value>(convert_a_key_to_a_value(realm, GC::Ref(*optional_key)));
+    });
```

Comment:
Mostly nit, but it removes a bit noise in my opinion
```suggestion
    auto operation = GC::Function<WebIDL::ExceptionOr<JS::Value>()>::create(realm.heap(), [&realm, &store, clone, key_value, no_overwrite] -> WebIDL::ExceptionOr<JS::Value> {
        auto maybe_key = store_a_record_into_an_object_store(realm, store, clone, key_value, no_overwrite);
        if (maybe_key.is_error())
            return WebIDL::ExceptionOr<JS::Value>(maybe_key.release_error());

        auto optional_key = maybe_key.release_value();
        if (optional_key == nullptr)
            return JS::js_undefined();

        return convert_a_key_to_a_value(realm, GC::Ref(*optional_key));
    });
```